clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -275,7 +275,8 @@ private RecordAppendResult tryAppend(long timestamp, byte[] key, byte[] value, H
     }
 
     private boolean isMuted(TopicPartition tp, long now) {
-        boolean result = muted.containsKey(tp) && muted.get(tp) > now;
+        Long throttleUntilTime = muted.get(tp);
+        boolean result = throttleUntilTime != null && throttleUntilTime > now;
         if (!result)
```

Comment:
This should probably be `throttleUntilTime != null && throttleUntilTime <= now` or something?